projects/plugins/jetpack/modules/subscriptions/subscribe-modal/class-jetpack-subscribe-modal.php

```diff
@@ -98,7 +109,15 @@ public function get_block_template_filter( $block_template, $id, $template_type
 	 */
 	public function get_block_templates_filter( $query_result, $query, $template_type ) {
 		if ( empty( $query ) && $template_type === 'wp_template_part' ) {
-			$query_result[] = $this->get_custom_template();
+			if ( is_array( $query_result ) ) {
+				// find the custom template and return early if we have a custom version in the results.
+				foreach ( $query_result as $template ) {
+					if ( $template->id === self::get_block_template_part_id() ) {
+						return $query_result;
+					}
+				}
+			}
+			$query_result[] = $this->get_template();
```

Comment:
Out of curiosity, can you give a quick note for why this change was needed? Just a bit hard for me to follow.

Reply:
Good call! something I didn't mentioned on the PR description. Without this PR when switching between areas "Headers", "Footers", "Uncategorized" the template is being added again and again to the lists.